core/api/src/main/java/com/alipay/sofa/rpc/context/RpcInvokeContext.java

```diff
@@ -59,6 +96,45 @@ public static RpcInvokeContext getContext() {
         return context;
     }
 
+    /**
+     * 设置上下文
+     *
+     * @param context 调用上下文
+     */
+    public static void setContext(RpcInvokeContext context) {
+        LOCAL.set(RpcInvokeContext.clone(context));
+    }
+
+    private static RpcInvokeContext clone(RpcInvokeContext parent) {
+        if (parent == null) {
+            return null;
+        }
+        RpcInvokeContext child = new RpcInvokeContext();
+        //timeout
+        child.setTimeout(parent.getTimeout());
+        //targetURL
+        child.setTargetURL(parent.getTargetURL());
+        //targetGroup
+        child.setTargetGroup(parent.getTargetGroup());
+        //responseCallback
+        child.setResponseCallback(parent.getResponseCallback());
+        //future
+        child.setFuture(parent.getFuture());
+        //map
+        child.map.putAll(parent.map);
+        //customHeader
+        child.customHeader.putAll(parent.customHeader);
+
+        if (BAGGAGE_ENABLE) {
```

Comment:
The clone method only needs to copy according to the parent. Should there judge according to the BAGGAGE_ENABLE.